### PR TITLE
PT-1004: Document the api using openApi and swaggo

### PR DESCRIPTION
**Pull request by yvon-dblg**
Only platform-data is documented

### CODE REVIEW

File: doc.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh -eu
```
Comment (BigFab) [inline doc.sh]: Can we name this one "buildDoc.sh" ?

**Review by BigFab — DISMISSED** [review]
Except for the few changes that do not relate to openapi documentation, I am fine with this. I can generate the swagger doc locally

**Review by BigFab — CHANGES_REQUESTED** [review]
Can you please rename doc.sh to buildDoc.sh?

**Comment by yvon-dblg** [comment]
> 
> 
> Can you please rename doc.sh to buildDoc.sh?

yes. I used this name, because for hydrophone (https://github.com/mdblp/hydrophone/blob/dblp/doc.sh) it was the one used.

Should we have something in common?
Maybe via an artifact ?

**Comment by yvon-dblg** [comment]
> 
> 
> Except for the few changes that do not relate to openapi documentation, I am fine with this. I can generate the swagger doc locally

ha yes, I commit my change in the README... sorry.
I will remove it.

**Review by BigFab — APPROVED** [review]
LGTM. Now I understand what these "enums" thing are here for. Thanks @yvon-dblg

**Pull request merged**